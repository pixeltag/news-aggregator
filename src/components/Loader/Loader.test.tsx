import { render, screen, cleanup } from '@testing-library/react';
import { Loader } from './Loader';

test('Should render Loader component', () => {
    render(<Loader />);
    const loaderElement = screen.getByTestId('loader');
    expect(loaderElement).toBeInTheDocument();
})


test('It renders correctly', () => {
    const element = render(<Loader />)
    expect(element).toMatchSnapshot();
})