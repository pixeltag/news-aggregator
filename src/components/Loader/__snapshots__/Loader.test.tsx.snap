// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="spinner-container"
        data-testid="loader"
      >
        <svg
          height="48"
          id="svg-spinner"
          viewBox="0 0 48 48"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="24"
            cy="4"
            fill="#fff"
            r="4"
          />
          <circle
            cx="12.19"
            cy="7.86"
            fill="#fffbf2"
            r="3.7"
          />
          <circle
            cx="5.02"
            cy="17.68"
            fill="#fef7e4"
            r="3.4"
          />
          <circle
            cx="5.02"
            cy="30.32"
            fill="#fef3d7"
            r="3.1"
          />
          <circle
            cx="12.19"
            cy="40.14"
            fill="#feefc9"
            r="2.8"
          />
          <circle
            cx="24"
            cy="44"
            fill="#feebbc"
            r="2.5"
          />
          <circle
            cx="35.81"
            cy="40.14"
            fill="#fde7af"
            r="2.2"
          />
          <circle
            cx="42.98"
            cy="30.32"
            fill="#fde3a1"
            r="1.9"
          />
          <circle
            cx="42.98"
            cy="17.68"
            fill="#fddf94"
            r="1.6"
          />
          <circle
            cx="35.81"
            cy="7.86"
            fill="#fcdb86"
            r="1.3"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="spinner-container"
      data-testid="loader"
    >
      <svg
        height="48"
        id="svg-spinner"
        viewBox="0 0 48 48"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="24"
          cy="4"
          fill="#fff"
          r="4"
        />
        <circle
          cx="12.19"
          cy="7.86"
          fill="#fffbf2"
          r="3.7"
        />
        <circle
          cx="5.02"
          cy="17.68"
          fill="#fef7e4"
          r="3.4"
        />
        <circle
          cx="5.02"
          cy="30.32"
          fill="#fef3d7"
          r="3.1"
        />
        <circle
          cx="12.19"
          cy="40.14"
          fill="#feefc9"
          r="2.8"
        />
        <circle
          cx="24"
          cy="44"
          fill="#feebbc"
          r="2.5"
        />
        <circle
          cx="35.81"
          cy="40.14"
          fill="#fde7af"
          r="2.2"
        />
        <circle
          cx="42.98"
          cy="30.32"
          fill="#fde3a1"
          r="1.9"
        />
        <circle
          cx="42.98"
          cy="17.68"
          fill="#fddf94"
          r="1.6"
        />
        <circle
          cx="35.81"
          cy="7.86"
          fill="#fcdb86"
          r="1.3"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
